client/lib/plugins/store.js

```diff
@@ -314,7 +314,8 @@ PluginsStore.dispatchToken = Dispatcher.register( function( payload ) {
 	switch ( action.type ) {
 		case 'RECEIVE_PLUGINS':
 			if ( action.error ) {
-				// bail if there is an error since there will be nothing to update
+				updatePlugins( action.site, [] );
```

Comment:
because we want to be able to know that the call has been done! If we keep it as undefined (vs an empty array), we would be still trying to fetch the plugins next time we try to get them from the store. If we want to avoid storing an empty array in this case, we would need to track if the data has been previously requested or not in another dictionary of sites... And right now I don't see any pro to that approach